backend/services/implementations/testSessionService.ts

```diff
@@ -57,6 +57,30 @@ class TestSessionService implements ITestSessionService {
     return testSessionDtos;
   }
 
+  async getTestSessionsByTeacherId(
+    teacherId: string,
+  ): Promise<Array<TestSessionResponseDTO>> {
+    let testSessionDtos: Array<TestSession> | null;
+
+    try {
+      testSessionDtos = await MgTestSession.find({
+        teacher: { $eq: teacherId },
+      });
+
+      // check if no testSessions are associated with the given teacherId
+      if (!testSessionDtos || !testSessionDtos.length) {
```

Comment:
why can't we just check if the length == 0?

Reply:
now that I think about this some more - does it make sense to throw an error here. Why not just return an empty array?

Reply:
hm i think your point makes sense to return an empty array - will update the code!